Source/WebCore/style/values/backgrounds/StyleBackgroundLayer.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 1999 Antti Koivisto (koivisto@kde.org)
+ * Copyright (C) 2004, 2005, 2006, 2007, 2008, 2014 Apple Inc. All rights reserved.
+ * Copyright (C) 2025 Samuel Weinig <sam@webkit.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "config.h"
+#include "StyleBackgroundLayer.h"
+
+#include "CachedImage.h"
+#include "StylePrimitiveNumericTypes+Logging.h"
+#include <wtf/PointerComparison.h>
+#include <wtf/TZoneMallocInlines.h>
+#include <wtf/text/TextStream.h>
+
+namespace WebCore {
+namespace Style {
+
+BackgroundLayer::BackgroundLayer(ImageOrNone&& image)
+    : m_image(WTFMove(image))
+    , m_position({ BackgroundLayer::initialFillXPosition(), BackgroundLayer::initialFillYPosition() })
+    , m_size(BackgroundLayer::initialFillSize())
+    , m_repeat(BackgroundLayer::initialFillRepeat())
+    , m_attachment(static_cast<unsigned>(BackgroundLayer::initialFillAttachment()))
+    , m_clip(static_cast<unsigned>(BackgroundLayer::initialFillClip()))
+    , m_origin(static_cast<unsigned>(BackgroundLayer::initialFillOrigin()))
+    , m_blendMode(static_cast<unsigned>(BackgroundLayer::initialFillBlendMode()))
+{
+}
+
+BackgroundLayer::BackgroundLayer(CSS::Keyword::None)
+    : BackgroundLayer { ImageOrNone { CSS::Keyword::None { } } }
+{
+}
+
+BackgroundLayer::BackgroundLayer(RefPtr<StyleImage>&& image)
+    : BackgroundLayer { ImageOrNone { WTFMove(image) } }
+{
+}
+
+bool BackgroundLayer::operator==(const BackgroundLayer& o) const
+{
+    return m_image == o.m_image
+        && m_position == o.m_position
+        && m_attachment == o.m_attachment
+        && m_clip == o.m_clip
+        && m_blendMode == o.m_blendMode
+        && m_origin == o.m_origin
+        && m_repeat == o.m_repeat
+        && m_size == o.m_size;
+}
```

Comment:
Can we use the default instead?

Reply:
No, we don’t want to include m_clipMax. 

Will add a comment to make that reasoning clear.